src/System.Text.Utf8/tests/TypeConstraintsTests.cs

```diff
@@ -0,0 +1,46 @@
+﻿using System.Reflection;
+using Xunit;
+
+namespace System.Text.Utf8.Tests
+{
+    public class TypeConstraintsTests
+    {
+        [Fact]
+        public void Utf8StringIsAStruct()
+        {
+            var utf8String = "anyString"u8;
```

Comment:
I am just proposing you remove the repeated code.

You can do something like 

private readonly Utf8String _utf8String = "anyString"u8;

and then, in your tests you just do

Assert(_utf8String.GetType()...).

This is not another abstraction layer and I am not proposing or trying to out optimize the compiler. I am optimizing for readability and for removal of duplicate code.